pgjdbc/src/test/java/org/postgresql/test/TestUtil.java

```diff
@@ -971,50 +973,88 @@ public static String queryForString(Connection conn, String sql) throws SQLExcep
    * Retrieve the backend process id for a given connection.
    */
   public static int getBackendPid(Connection conn) throws SQLException {
-    Statement stmt = conn.createStatement();
-    ResultSet rs = stmt.executeQuery("SELECT pg_backend_pid()");
-    rs.next();
-    int pid = rs.getInt(1);
-    rs.close();
-    stmt.close();
-    return pid;
+    PGConnection pgConn = conn.unwrap(PGConnection.class);
+    return pgConn.getBackendPID();
+  }
+
+  public static boolean isPidAlive(Connection conn, int pid) throws SQLException {
+    String sql = "SELECT EXISTS (SELECT * FROM pg_stat_activity WHERE pid = ?) t";
```

Comment:
seems pid is not available on 8.4. We can probably remove this test from 8.4

Reply:
There's `procpid` column that could be used instead. I'll put an if / else on the backend version.

I took a look and the `procpid` was renamed to `pid` in 9.2 so this would have failed for 9.0 and 9.1 as well but we don't have the min our matrix anymore.